packages/engine/src/worker/runner/python/state.py

```diff
@@ -92,32 +105,39 @@ def modify(self, field_name, fn):
 
     # `data` is an optional argument. `data` must be JSON-serializable.
     def add_message(self, to, msg_type, data=None):
-        idx = self.__idx
-        self.__dict__['__msgs'][idx].append({
-            "to": [to] if isinstance(to, str) else to,
-            "type": msg_type,
-            "data": deepcopy(data) if self.__dict__['__msgs_native'][idx] else json.dumps(data)
-        })
-
-    # Returns the index of the currently executing behavior in the agent's behavior chain.
+        idx = self.__dict__["__idx_in_group"]
+        self.__dict__["__msgs"][idx].append(
+            {
+                "to": [to] if isinstance(to, str) else to,
+                "type": msg_type,
+                "data": deepcopy(data)
```

Comment:
Was this pylint? Having the if else broken up makes it pretty unpleasant. Maybe we could just move it above into `data_as_json = deepcopy(data) if self.__dict__["__msgs_native"][idx] else json.dumps(data)`

Although I guess Pylint might break it up again, but at least it's not within a dict definition